### PR TITLE
SEO: Move to free

### LINKED ISSUE #13827: SEO Tools: add fields for custom title and description to post editor
**What's the feature I'm requesting?**
Ability to override SEO title/description tags (independently of the actual post title/description).

**What I'm seeing**
I think we have some room for improvement in being able to modify the SEO title tag and meta description on a post level. Although you can set custom title formats by reordering Site Name/Date/Post Title/ect., there is some missed opportunity to modify those title tags beyond the post title (see: https://jetpack.com/support/seo-tools/).

Say for example that you have your title tag set to pull in the post title and site name and you want to target a particular keyword in your title tag, you would have to make your post title that keyword. This doesn't lend itself well to being able to search internally for posts, especially considering those who have a lot of content. The goal would be to be able to modify the title tag independently from the post title. 
Here's an example:
Screen Shot 2019-10-23 at 4 53 56 PM
Screen Shot 2019-10-23 at 4 45 10 PM

So if I want to optimize my title tag for the search term `Best Type of Potato` with my website placed at the end of that tile tag, this is what it would look like:
Screen Shot 2019-10-23 at 4 52 42 PM
Screen Shot 2019-10-23 at 4 53 23 PM

As far as meta descriptions go, you can currently set a meta description for the front page, but it would be good to be able to set a unique meta description within each post in order to craft a unique call to action. 

**Why am I requesting this?**
As a specialist in SEO, I need to be able to modify title tags and meta descriptions for every page and post on our site. There are unique keyword opportunities that we try to target, and providing a unique and concise title and meta description for each post/page helps us target those keywords on every page to follow on-page SEO best practices. To that length, posts/pages with duplicate or missing title tags and descriptions make it difficult for search engine crawlers to understand the page content, thus leading to a lower chance of ranking well.

**What's the ideal fix here?**
- Make a separate SEO title tag field, which is a feature that I believe is currently missing
- Make a separate SEO meta description field, which works at WordPress.com already, but ideally would make it work in the new block editor, also adding it to Jetpack self-hosted installs.

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes #13827 , p1HpG7-agp-p2

#### Changes proposed in this Pull Request:
* Make the SEO panels available for all sites.

#### Jetpack product discussion

* PT: p1HpG7-ba8-p2
* p1HpG7-agp-p2

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
* Connected Jetpack site
* Can you enable SEO without a plan? Does the React admin page allow you to activate it? No upsells?
* On a new post, is there an SEO panel in the block editor for modifying these fields?

#### Proposed changelog entry for your changes:
* SEO Tools: Make available for all sites.

### CODE REVIEW

**Comment by samiff** [comment]
This is testing well for me using the following steps:

**Testing instructions**:

* After pulling down these changes, connect Jetpack to WordPress.com using Jetpack Free as the plan.
* At no point in testing should you see nudges about upgrading for the SEO Tools features.
* On the `Traffic` settings tab `/wp-admin/admin.php?page=jetpack#/traffic` make sure you see a `Customize your SEO settings` link instead of an upgrade prompt under the "Search engine optimization" section.
* On the `My Plan` page `/wp-admin/admin.php?page=jetpack#/my-plan` make sure that you see the `Activate SEO Tools` card.
* Activate the SEO Tools module from the `My Plan` page, and check `/wp-admin/admin.php?page=jetpack_modules` to make sure the module was activated ok.
* Open a post for editing and make sure you can edit/save the post's custom SEO Description:

Markup on 2021-02-08 at 15:17:30

---

I've also tested with the yet to be merged Calypso [PR](https://github.com/Automattic/wp-calypso/pull/49504) and that was working great for me as well, allowing control over the SEO Tools features on a free JP connected site.

**Comment by samiff** [comment]
D50386-code should be up to date now, and I've tested those changes again. I also tried a free WPcom simple site with both the JP & Calypso changes applied, and the upgrade nudge was shown which is the expected.

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/traffic/seo.jsx:62]: Could we allow one to activate the feature here, if it's not already active, and only show the link to customize once the module is active? 
This is what we do for the Sharing or Publicize modules for example, or on the "My Plan" page for the SEO tools.

File: projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php:48]: I believe you may be able to refactor `is_enabled_jetpack_seo` since we don't need to check for a plan anymore. We may even be able to get rid of it entirely alongside the `jetpack_disable_seo_tools` filter, and instead ensure that everything is shortcircuited early when the module is not active or when a conflicting plugin is installed (see `projects/plugins/jetpack/modules/seo-tools.php`).

File: projects/plugins/jetpack/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js:850]: You should now be able to get rid of all the plan checking logic above this.

**Review by jeherve — CHANGES_REQUESTED** [review]
I left a few remarks below.

*********

It may be worth checking all other files to ensure the feature is not linked to plans anywhere else, such as here:
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/_inc/client/lib/plans/constants.js#L172

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx:62]: That would be a nice change, though it seems like the ideal goal would be to have the meta fields available in wp-admin, which would also work well if we want this to work for a user-less connection. I'll need to explore this a bit more to see what implementing the controls to wp-admin would look like.

Comment (jeherve) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx:62]: Sure thing. I think adding those fields to wp-admin is a bit beyond the scope of this PR though. I would personally recommend that we focus on the move to free in this PR, and then develop the new interface to manage settings in wp-admin in a follow-up PR.

File: projects/plugins/jetpack/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js:831]: 89d17d72387312ae325a547c78dcf7e0a4c5d922

File: projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php
Comment (samiff) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: @jeherve Nice catch, from what I tested the plan check is not needed in `is_enabled_jetpack_seo()` so it's been removed in https://github.com/Automattic/jetpack/pull/17309/commits/073a7385b98951936c1f85499fef0550bc0e0075

> We may even be able to get rid of it entirely alongside the jetpack_disable_seo_tools filter...

I'm a bit more hesitant with making this change, as the filter comment mentions:

> Can be used by SEO plugin authors to disable the conflicting output of SEO Tools.

I'm not sure of the likelihood of that actually being used, and don't think it would hurt leaving it in, but I'll defer to your expertise on this.

Comment (jeherve) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: Yeah, we could keep the filter for a bit longer. Maybe we could move it to `projects/plugins/jetpack/modules/seo-tools.php` instead of having `is_enabled_jetpack_seo()`, and shortcircuit things early (e.g. not even including `seo-tools/jetpack-seo.php`) when the filter is in use, or when a conflicting plugin is installed.

Comment (samiff) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: I see the static method in places like:

https://github.com/Automattic/jetpack/blob/222e0ec1fb9b7984bbcb0d4b759c457961d9bc1b/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php#L795

Which is called on the Calypso side when saving a new front page meta description for example. Should we leave that is for now? I did add the filter check in the `seo-tools.php` file in https://github.com/Automattic/jetpack/pull/17309/commits/3a5da042409614a7b47bc461d63e163441263816  that look ok?

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx:60]: Tracking the WP Admin portion in: https://github.com/Automattic/jetpack/pull/18810

Comment (jeherve) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx:60]: I still think we should allow one to activate the module from here now. In the follow-up PR we can expand on things and have the additional settings show up instead of just the link to Calypso, but until then, it would be nice if one were able to activate the module here when it's inactive, instead of having to go to the My Plan page (where I wouldn't think to go when I have a free plan, so I think we could even remove the card from there at this point).

This would also allow us to get rid of the logic with `getModuleOverridenBanner`, since the module would use a more traditional module toggle.

Something like this?

```diff
diff --git a/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx b/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
index 7964e86de..a64ac52d0 100644
--- a/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
+++ b/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx
@@ -15,7 +15,6 @@ import Button from 'components/button';
 import {
 	FEATURE_SECURITY_SCANNING_JETPACK,
 	FEATURE_SITE_BACKUPS_JETPACK,
-	FEATURE_SEO_TOOLS_JETPACK,
 	FEATURE_VIDEO_HOSTING_JETPACK,
 	FEATURE_GOOGLE_ANALYTICS_JETPACK,
 	FEATURE_WORDADS_JETPACK,
@@ -299,8 +298,6 @@ export const SettingsCard = props => {
 				return 'inactive' === props.getModuleOverride( 'wordads' );
 			case FEATURE_GOOGLE_ANALYTICS_JETPACK:
 				return 'inactive' === props.getModuleOverride( 'google-analytics' );
-			case FEATURE_SEO_TOOLS_JETPACK:
-				return 'inactive' === props.getModuleOverride( 'seo-tools' );
 			case FEATURE_SEARCH_JETPACK:
 				return 'inactive' === props.getModuleOverride( 'search' );
 			default:
@@ -308,22 +305,18 @@ export const SettingsCard = props => {
 		}
 	};
 
-	// We only want to show this banner for Google Analytics and SEO Tools because
+	// We only want to show this banner for Google Analytics because
 	// they don't use the ModuleToggle for their UI.
-	const getModuleOverridenBanner = () => {
+	const getGoogleAnalyticsOverridenBanner = () => {
 		if ( ! featureIsOverriden() ) {
 			return false;
 		}
-		switch ( feature ) {
-			case FEATURE_GOOGLE_ANALYTICS_JETPACK:
-				const googleAnalytics = props.getModule( 'google-analytics' );
-				return <ModuleOverridenBanner moduleName={ googleAnalytics.name } />;
-			case FEATURE_SEO_TOOLS_JETPACK:
-				const seoTools = props.getModule( 'seo-tools' );
-				return <ModuleOverridenBanner moduleName={ seoTools.name } />;
-			default:
-				return null;
+		if ( feature !== FEATURE_GOOGLE_ANALYTICS_JETPACK ) {
+			return null;
 		}
+
+		const googleAnalytics = props.getModule( 'google-analytics' );
+		return <ModuleOverridenBanner moduleName={ googleAnalytics.name } />;
 	};
 
 	const children = showChildren() && props.children;
@@ -342,7 +335,7 @@ export const SettingsCard = props => {
 	}
 
 	return (
-		getModuleOverridenBanner() || (
+		getGoogleAnalyticsOverridenBanner() || (
 			<form
 				{ ...( moduleId ? { id: moduleId } : null ) }
 				className={ `jp-form-settings-card` }
@@ -398,7 +391,6 @@ export default connect( state => {
 		securityProUpgradeUrl: getUpgradeUrl( state, 'settings-security-pro' ),
 		securityPremiumUpgradeUrl: getUpgradeUrl( state, 'settings-security-premium' ),
 		gaUpgradeUrl: getUpgradeUrl( state, 'settings-ga' ),
-		seoUpgradeUrl: getUpgradeUrl( state, 'settings-seo' ),
 		searchUpgradeUrl: getUpgradeUrl( state, 'jetpack-search' ),
 		spamUpgradeUrl: getUpgradeUrl( state, 'settings-spam' ),
 		multisite: isMultisite( state ),
diff --git a/projects/plugins/jetpack/_inc/client/lib/plans/constants.js b/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
index dddae6dbe..e151ae0b1 100644
--- a/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
+++ b/projects/plugins/jetpack/_inc/client/lib/plans/constants.js
@@ -154,7 +154,6 @@ export const FEATURE_REVENUE_GENERATION_JETPACK = 'revenue-generation-jetpack';
 export const FEATURE_VIDEO_HOSTING_JETPACK = 'video-hosting-jetpack';
 export const FEATURE_SECURITY_ESSENTIALS_JETPACK = 'security-essentials-jetpack';
 export const FEATURE_PRIORITY_SUPPORT_JETPACK = 'priority-support-jetpack';
-export const FEATURE_SEO_TOOLS_JETPACK = 'seo-tools-jetpack';
 export const FEATURE_WORDADS_JETPACK = 'wordads-jetpack';
 export const FEATURE_GOOGLE_ANALYTICS_JETPACK = 'google-analytics-jetpack';
 export const FEATURE_SEARCH_JETPACK = 'search-jetpack';
diff --git a/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx b/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
index 54ca777e1..19cd23e5d 100644
--- a/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
+++ b/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx
@@ -68,56 +68,11 @@ class MyPlanBody extends React.Component {
 		this.trackPlansClick( 'activate_videopress' );
 	};
 
-	activateSeo = () => {
-		this.props.activateModule( 'seo-tools' );
-		this.trackPlansClick( 'activate_seo' );
-	};
-
 	activateGoogleAnalytics = () => {
 		this.props.activateModule( 'google-analytics' );
 		this.trackPlansClick( 'activate_ga' );
 	};
 
-	getSeoToolsCard = () => {
-		return (
-			<div className="jp-landing__plan-features-card">
-				<div className="jp-landing__plan-features-img">
-					<img
-						src={ imagePath + '/jetpack-performance-icon.svg' }
-						className="jp-landing__plan-features-icon"
-						alt={ __( 'Site stats showing an evolution in traffic and engagement', 'jetpack' ) }
-					/>
-				</div>
-				<div className="jp-landing__plan-features-text">
-					<h3 className="jp-landing__plan-features-title">{ __( 'SEO Tools', 'jetpack' ) }</h3>
-					<p>
-						{ __(
-							'Advanced SEO tools to help your site get found when people search for relevant content.',
-							'jetpack'
-						) }
-					</p>
-					{ this.props.isModuleActivated( 'seo-tools' ) ? (
-						<Button
-							onClick={ this.handleButtonClickForTracking( 'configure_seo' ) }
-							href={ getRedirectUrl( 'calypso-marketing-traffic', {
-								site: this.props.siteRawUrl,
-							} ) }
-						>
-							{ __( 'Configure site SEO', 'jetpack' ) }
-						</Button>
-					) : (
-						<Button
-							onClick={ this.activateSeo }
-							disabled={ this.props.isActivatingModule( 'seo-tools' ) }
-						>
-							{ __( 'Activate SEO tools', 'jetpack' ) }
-						</Button>
-					) }
-				</div>
-			</div>
-		);
-	};
-
 	render() {
 		let planCard = '';
 		const planClass = 'offline' !== this.props.plan ? getPlanClass( this.props.plan ) : 'offline';
@@ -502,8 +457,6 @@ class MyPlanBody extends React.Component {
 							</div>
 						) }
 
-						{ 'inactive' !== this.props.getModuleOverride( 'seo-tools' ) && this.getSeoToolsCard() }
-
 						{ isPlanPremiumOrBetter &&
 							'inactive' !== this.props.getModuleOverride( 'google-analytics' ) && (
 								<div className="jp-landing__plan-features-card">
@@ -757,8 +710,6 @@ class MyPlanBody extends React.Component {
 							) }
 						</div>
 
-						{ 'inactive' !== this.props.getModuleOverride( 'seo-tools' ) && this.getSeoToolsCard() }
-
 						<div className="jp-landing__plan-features-card">
 							<div className="jp-landing__plan-features-img">
 								<img
diff --git a/projects/plugins/jetpack/_inc/client/traffic/seo.jsx b/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
index cacfb2a09..a5e78d050 100644
--- a/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
+++ b/projects/plugins/jetpack/_inc/client/traffic/seo.jsx
@@ -1,8 +1,7 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { Component } from 'react';
 import { __, _x } from '@wordpress/i18n';
 
 /**
@@ -11,60 +10,66 @@ import { __, _x } from '@wordpress/i18n';
 import analytics from 'lib/analytics';
 import Card from 'components/card';
 import getRedirectUrl from 'lib/jp-redirect';
-import { FEATURE_SEO_TOOLS_JETPACK } from 'lib/plans/constants';
+import { ModuleToggle } from 'components/module-toggle';
 import { withModuleSettingsFormHelpers } from 'components/module-settings/with-module-settings-form-helpers';
 import SettingsCard from 'components/settings-card';
 import SettingsGroup from 'components/settings-group';
-import { getSitePlan } from 'state/site';
 
-class SeoComponent extends React.Component {
-	trackConfigureClick = () => {
-		analytics.tracks.recordJetpackClick( 'configure-seo' );
-	};
+export const SEO = withModuleSettingsFormHelpers(
+	class extends Component {
+		trackConfigureClick = () => {
+			analytics.tracks.recordJetpackClick( 'configure-seo' );
+		};
 
-	render() {
-		return (
-			<SettingsCard
-				{ ...this.props }
-				header={ _x( 'Search engine optimization', 'Settings header', 'jetpack' ) }
-				feature={ FEATURE_SEO_TOOLS_JETPACK }
-				hideButton
-			>
-				<SettingsGroup
-					disableInOfflineMode
-					module={ { module: 'seo-tools' } }
-					support={ {
-						text: __(
-							'Allows you to optimize your site and its content for better results in search engines.',
-							'jetpack'
-						),
-						link: getRedirectUrl( 'jetpack-support-seo-tools' ),
-					} }
+		render() {
+			const isActive = this.props.getOptionValue( 'seo-tools' );
+
+			return (
+				<SettingsCard
+					{ ...this.props }
+					header={ _x( 'Search engine optimization', 'Settings header', 'jetpack' ) }
+					feature={ 'seo-tools-jetpack' }
+					module="seo-tools"
+					hideButton
 				>
-					<span>
-						{ __(
-							'Take control of the way search engines represent your site. With Jetpack’s SEO tools you can preview how your content will look on popular search engines and change items like your site name and tagline in seconds.',
-							'jetpack'
-						) }
-					</span>
-				</SettingsGroup>
-				{ ! this.props.isUnavailableInOfflineMode( 'seo-tools' ) && (
-					<Card
-						compact
-						className="jp-settings-card__configure-link"
-						onClick={ this.trackConfigureClick }
-						href={ this.props.configureUrl }
+					<SettingsGroup
+						disableInOfflineMode
+						module={ { module: 'seo-tools' } }
+						support={ {
+							text: __(
+								'Allows you to optimize your site and its content for better results in search engines.',
+								'jetpack'
+							),
+							link: getRedirectUrl( 'jetpack-support-seo-tools' ),
+						} }
 					>
-						{ __( 'Customize your SEO settings', 'jetpack' ) }
-					</Card>
-				) }
-			</SettingsCard>
-		);
+						<p>
+							{ __(
+								'Take control of the way search engines represent your site. With Jetpack’s SEO tools you can preview how your content will look on popular search engines and change items like your site name and tagline in seconds.',
+								'jetpack'
+							) }
+						</p>
+						<ModuleToggle
+							slug="seo-tools"
+							activated={ isActive }
+							toggling={ this.props.isSavingAnyOption( 'seo-tools' ) }
+							toggleModule={ this.props.toggleModuleNow }
+						>
+							{ __( 'Customize your SEO settings', 'jetpack' ) }
+						</ModuleToggle>
+					</SettingsGroup>
+					{ isActive && ! this.props.isOfflineMode && (
+						<Card
+							compact
+							className="jp-settings-card__configure-link"
+							onClick={ this.trackConfigureClick }
+							href={ this.props.configureUrl }
+						>
+							{ __( 'Customize your SEO settings', 'jetpack' ) }
+						</Card>
+					) }
+				</SettingsCard>
+			);
+		}
 	}
-}
-
-export const SEO = connect( state => {
-	return {
-		sitePlan: getSitePlan( state ),
-	};
-} )( withModuleSettingsFormHelpers( SeoComponent ) );
+);

```

**Review by jeherve — CHANGES_REQUESTED** [review]
I did another review, and left a few comments inline. Let me know what you think!

Also, as mentioned in the linting results, you should now be able to edit `tools/phpcs-excludelist.json` to remove `projects/plugins/jetpack/modules/seo-tools.php` since it's now error-free. 🥳

File: projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php
Comment (jeherve) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: Yeah 👍

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx:14]: Looks like it's also being used in this file twice:

https://github.com/Automattic/jetpack/blob/91c96b27af3d1e824c9c784e4dd3925e993f70fd/projects/plugins/jetpack/_inc/client/components/settings-card/index.jsx#L302

So I think leaving that in constants.js should be ok for now.

File: projects/plugins/jetpack/modules/seo-tools.php
Comment (samiff) [reply projects/plugins/jetpack/modules/seo-tools.php:42]: > I think you'll need to keep the require at the top of this file, as we need it above when we check for active plugins.

Do you mean just the require to `plugin.php` or both?

Also, could you help me understand this a bit better. I see the call to `Jetpack::is_plugin_active`, shouldn't this just work since the Jetpack plugin is already running at this point?

File: projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php
Comment (samiff) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: That makes sense, thanks for pointing it out! https://github.com/Automattic/jetpack/pull/17309/commits/6e70736e82c14f5bc1a1fb0915b5d8897cece928

File: projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php
Comment (samiff) [reply projects/plugins/jetpack/modules/seo-tools/jetpack-seo-utils.php]: That makes sense, the plan checking logic that was in `is_enabled_jetpack_seo` has been removed; removed in https://github.com/Automattic/jetpack/pull/17309/commits/6e70736e82c14f5bc1a1fb0915b5d8897cece928

File: projects/plugins/jetpack/_inc/client/traffic/seo.jsx
Comment (samiff) [reply projects/plugins/jetpack/_inc/client/traffic/seo.jsx]: Awesome! Looks good and test well on my end 👍 

* https://github.com/Automattic/jetpack/pull/17309/commits/2631a3c950fbe2c91d11d125f9ab273c8c196051
* https://github.com/Automattic/jetpack/pull/17309/commits/f0ec8fdb2ff20cc94b424cb43a1f8af9edf8eebb
* https://github.com/Automattic/jetpack/pull/17309/commits/a7c42c6c4ee9eaa88a982fd8cc0c3dc442a47125

File: projects/plugins/jetpack/modules/seo-tools.php
Comment (jeherve) [reply projects/plugins/jetpack/modules/seo-tools.php:42]: Just `plugin.php`.

> could you help me understand this a bit better. I see the call to Jetpack::is_plugin_active, shouldn't this just work since the Jetpack plugin is already running at this point?

I think it was added in 153691a3d8cf18e7b2c9f9160f696ad972003207 as a safeguard before calling core's `is_plugin_active()`, but you're right, it may indeed just work now that we use `Jetpack::is_plugin_active()`. You could remove it altogether now!

```suggestion
```

**Review by jeherve — CHANGES_REQUESTED** [review]
Apart from my comment above, this tests well and looks good to me! It should be good to merge after that last change.

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
I'll merge this now so you can continue to work on #18810.

**Comment by kraftbj** [comment]
r222059-wpcom

**Comment by kraftbj** [comment]
and r222084-wpcom

**Pull request merged**
